jep/0000/README.adoc

```diff
@@ -0,0 +1,220 @@
+the other side, the legacy constructor is still available (but deprecated) and do define a default value
+for this new argument.
+
+**Shall we just relax Configuration-as-Code so it still can apply an outdated model ?**
+
+We don't think so, as doing so would eventually bring us to indeterminable configuration paths:
+As we allow use of deprecated constructor (or setters, or ...) we might result with colliding candidates
+to invoke, without a deterministic way to select the adequate one.
+
+To follow-up with this specific sample, JNLPLauncher do define (at time writing) two deprecated constructors
+and another non-deprecated one :
+```java
+@Deprecated
+public JNLPLauncher(String tunnel, String vmargs) { ... }
+
+@Deprecated
+public JNLPLauncher() { ... }
+
+public JNLPLauncher(boolean enableWorkDir) { ... }
+
+```
+The sole one we can guarantee is designed to submit configuration is the @DataBound one. For others
+there's no deterministic way to choose one. Even considering the configuration attributes we have from
+yaml file, which one should we use if configuration doesn't define `enableWorkDir`, `tunnel` nor `vmargs`?
+first one passing `null, null` of non-deprecated one passing `false` default value ?
+
+=== During Jenkins development
+
+Actually the exact same code change discussed in https://github.com/jenkinsci/configuration-as-code-plugin/issues/233[issue 233]
+could have been implemented without breaking Configuration-as-Code inferred model :
+
+- by defining a `@DataBoundSetter` for newly introduced `workDirSettings` attribute
+- by not defining `workDirSettings` argument as `@Nonnull` and applying default value
+
+Assuming Configuration-as-Code would exist for some time when this change was applied to jenkins-core, we
+might have detected it during code review and prevented this to happen.
+
+Configuration-as-Code tooling wants to avoid such unnecessary breaking changes to happen in the future.
+But as it wasn't there before developers aren't aware of this constraint, and we can't just ask them
+to keep our use-case in mind any time they write some code (there's already enough backward compatibility
+issues for them to address).
+
+Configuration-as-Code tooling do define tooling to detect such breaking changes, and warn developer about
+them. This tooling do run the model inference logic and compare with previous state of the codebase. This
+will warn developer about breaking change being introduced for Configuration-as-Code users.
+
+Sometime such changes are actually required. For such situation we propose some metadata being added aside
+changes made to codebase so Configuration-as-Code can give end-user some hints on the break and how to fix
+it.
+
+=== Before applying configuration
+
+On end-user side we want ot offer a validation toolkit to be used when applying configuration on a jenkins
+instance. This toolkit do parse the yaml configuration, try to apply configuration on the inferred model,
```

Comment:
BTW please follow the formatting model of one physical (AsciiDoc) text line per sentence or major phrase. Easier to read diffs and so on.